pennylane_lightning/src/Gates.hpp

```diff
@@ -323,6 +323,23 @@ namespace Pennylane {
         void applyKernel(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices);
     };
 
+    class Unitary : public AbstractGate {
```

Comment:
For the existing gates we knew the matrix representation in advance (and could hardcode them). Down the line, would be nice to check that there's no major performance degradation if we have a circuit with several layers that apply the same `QubitUnitary`. In such a case, parameters will be passed for every `QubitUnitary` and lightning will consider each of them separately. 🤔